Annex60/Fluid/FMI/Adaptors/ThermalZone.mo

```diff
@@ -0,0 +1,380 @@
+          fillPattern=FillPattern.Solid,
+          borderPattern=BorderPattern.Raised),
+        Rectangle(
+          extent={{-92,30},{100,-44}},
+          lineColor={0,0,0},
+          lineThickness=5.0,
+          fillColor={235,235,235},
+          fillPattern=FillPattern.Solid,
+          borderPattern=BorderPattern.Raised),
+        Text(
+          extent={{-96,15},{96,-15}},
+          lineColor={0,0,0},
+          textString="%y"),
+        Text(
+          extent={{-150,90},{140,50}},
+          textString="%name",
+          lineColor={0,0,255})}), Documentation(info="<html>
+<p>
+The (time varying) vector <code>Real</code> output signal of this block can be defined in its
+parameter menu via variable <code>y</code>. The purpose is to support the
+easy definition of vector-valued Real expressions in a block diagram.
+</p>
+</html>"));
+  end RealVectorExpression;
+
+  block x_i_toX_w "Conversion from Xi to X"
+    extends Modelica.Blocks.Icons.Block;
+
+    replaceable package Medium =
+      Modelica.Media.Interfaces.PartialMedium "Medium model within the source"
+     annotation (choicesAllMatching=true);
+    Modelica.Blocks.Interfaces.RealInput Xi[Medium.nXi](each final unit="kg/kg") if
+      Medium.nXi > 0 "Water vapor concentration in kg/kg total air"
+      annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+          iconTransformation(extent={{-140,-20},{-100,20}})));
+
+    Modelica.Blocks.Interfaces.RealOutput X_w(
+      final unit="kg/kg") "Water vapor concentration in kg/kg total air"
+      annotation (Placement(transformation(extent={{100,-20},{140,20}})));
+  protected
+    Modelica.Blocks.Interfaces.RealInput Xi_internal[Medium.nXi](
+      each final unit = "kg/kg")
+      "Internal connector for water vapor concentration in kg/kg total air";
+
+    Modelica.Blocks.Interfaces.RealInput X_w_internal(
+      final unit = "kg/kg")
+      "Internal connector for water vapor concentration in kg/kg total air";
+
+  equation
+  // Conditional connector
+  connect(Xi_internal, Xi);
+  if Medium.nXi == 0 then
+    Xi_internal = zeros(Medium.nXi);
+  end if;
+
+  X_w_internal = sum(Xi_internal);
+  connect( X_w, X_w_internal)
+  annotation (Documentation(revisions="<html>
+<ul>
+<li>
+November 8, 2016, by Michael Wetter:<br/>
+Removed wrong usage of <code>each</code> keyword.
+</li>
+<li>
+April 27, 2016, by Thierry S. Nouidui Wetter:<br/>
+First implementation.
+</li>
+</ul>
+</html>", info="<html>
+<p>
+Block that converts a vector input for the water mass fraction <code>Xi</code>
+to a scalar output <code>X</code>.
+This is needed for models in which a scalar input signal <code>Xi</code> that
+may be conditionally removed is to be connected to a model with a vector
+input <code>X</code>, because the conversion from scalar to vector
+needs to access the conditional connector, but conditional connectors
+can only be used in <code>connect</code> statements.
+</p>
+</html>"));
```

Comment:
@mwetter 
I meant to suggest to have all the documentation for the top-level model and the internal blocks in one place at the top-level, as some users may overlook the internal block documentation when they read the documentation of the top-level model. But having a second look at it, I also see your point in putting the further information to the internal blocks, keeping the top-level documentation easier to read. I'll merge as is.